### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by lnchambers**

### CODE REVIEW

File: lib/sales_engine.rb [hunk 1]
```diff
@@ -30,7 +32,9 @@ def self.from_csv(data)
   end
 
   def get_data
-    @merchant_data = merchants.get_revenue
+    @transaction_data   = transactions.successful_transactions
+    @invoice_item_data  = invoice_items.total(@transaction_data.keys)
+    @merchant_data      = merchants.get_revenue
   end
```
Comment (zpalmquist) [inline lib/sales_engine.rb:38]: maybe just break these out into methods?
def transaction_data
    transactions.successful_transactions
 end

etc

Comment (zpalmquist) [reply lib/sales_engine.rb:38]: the code thing on github is surprisingly horrible

File: lib/sales_engine.rb [hunk 2]
```diff
@@ -13,7 +13,9 @@ class SalesEngine
               :invoice_items,
               :customers,
               :transactions,
-              :merchant_data
+              :merchant_data,
+              :transaction_data,
+              :invoice_item_data
```
Comment (zpalmquist) [inline lib/sales_engine.rb:19]: can already hear them now.... see if we can move these out.

File: lib/sales_analyst.rb [hunk 3]
```diff
@@ -52,11 +52,6 @@ def average_average_price_per_merchant
     (total / merchants.all.count).round(2)
   end
 
-  def calculate_stdev(mean)
-    variance = mean.sum / (mean.count - 1)
-    Math.sqrt(variance).round(2)
-  end
-
```
Comment (zpalmquist) [inline lib/sales_analyst.rb:59]: where did they go?

File: lib/sales_analyst.rb [hunk 4]
```diff
@@ -101,18 +96,6 @@ def total_revenue_by_date(date)
...
-      invoice.created_at.to_i == date.to_i
-    end
-  end
-
-  def get_invoice_items_for_revenue(date)
-    get_invoices_for_revenue(date).map do |invoice|
-      @sales_engine.invoice_items.find_all_by_invoice_id(invoice.id)
-    end
-  end
-
```
Comment (zpalmquist) [inline lib/sales_analyst.rb:115]: assume this was moved out to sales_enginey things?

**Review by zpalmquist — CHANGES_REQUESTED** [review]

File: lib/sales_analyst.rb (hunk 3 above)
Comment (lnchambers) [reply lib/sales_analyst.rb:59]: They went to the merchant_analyst module

File: lib/sales_engine.rb (hunk 2 above)
Comment (lnchambers) [reply lib/sales_engine.rb:19]: I could make manual attr_readers for them

File: lib/sales_engine.rb (hunk 1 above)
Comment (lnchambers) [reply lib/sales_engine.rb:38]: If I did that, they would go through all their enumerables again =/

File: lib/sales_analyst.rb (hunk 4 above)
Comment (lnchambers) [reply lib/sales_analyst.rb:115]: It was moved from sales_analyst to invoice_analyst module

File: lib/sales_engine.rb (hunk 1 above)
Comment (zpalmquist) [reply lib/sales_engine.rb:38]: what if they were memoized?

Comment (lnchambers) [reply lib/sales_engine.rb:38]: I'll give it a shot =)

**Pull request closed without merging**